observability/microservices.md

```diff
@@ -0,0 +1,53 @@
+# Observability in Microservices
+
+Microservices is a very popular software architecture, where the application is arranged as a collection of loosely coupled services. Some of those services can be written in different languages by different teams.
+
+## Motivations
+
+We need to consider special cases when creating a microservice architecture from the perspective of observability. We want to capture the interactions when making requests between those microservices and correlate them.
+
+Imagine we have a microservice that accesses a database to retrieve some data as part of a request. This microservice is going to be called by someone else as part of an incoming http request or an internal process being executed. What happens if a problem occurs during the retrieval of the data (or the update of the data)? How can we associate, or correlate, that this particular call failed in the destination microservice?
+
+This is a common issue. When calling other microservices, depending on the technology stack we use, we can accidentally hide errors and exceptions that might happen on the other side. If we are using a simple REST interface, the other microservice can return a 500 HTTP status code and we don't have any idea what happen inside that microservice.
+
+More important, we don't have any way to associate our Correlation Id to whatever happens inside that microservice. Therefore, is so important to have a plan in place to be able to extend your traceability and monitoring efforts, especially when using a microservice architecture.
+
+## How to extend your tracing information between microservices
+
+The W3C consortium is working on a [Trace Context](https://www.w3.org/TR/trace-context/) definition that can be applied when using HTTP as the protocol in a microservice architecture. But let's explain how we can implement this functionality in our software.
+
+The main idea behind this is to propagate the correlation information between HTTP request so other pieces of software can read this information and correctly correlate telemetry across microservices.
+
+The way to propagate this information is to use HTTP Headers for the Correlation Id, parent Correlation Id, etc.
```

Comment:
I love this information and it is very nicely summarised. Maybe it is worth mentioning that most libraries that support distributed tracing already implement this for you. That way we can keep this practical for people.

Reply:
Yes, it's mention at the end of the post.

Reply:
Yeah, what I meant was providing a "TL;DR;" earlier and then the extended version in case the person reading would like to understand better how it works.

Reply:
Well I don't like the TLDR section, it just a page long markdown document, I don't think it make sense, but happy to do what you think is best.

Reply:
I agree with @ivaramme with this one, I;ve added a "Recommendation" snippet above for this